Michael1

michael9j2lee opened the pull request:
Please Review these changes.

michael9j2lee committed:
1. Add Thyme Leaf
2. Add new CityController.java
3. Add new templates/weather_show.html
4. Modify CityRestController.java

michael9j2lee committed:
Modified CityRestController.java to return new CityInfo
Modified City.java to only have city and country information
Modified weatherService to return TempAndTime
Modified CityRepository to return City information
Modified weather_show to return the new city information for the CityController.java

Added CityController to return weather_show with CityInfo.java
Added CityService To return the CityInfo Object with City and tempandTime
Added TempAndTime to hold only the time and temperature of City
Added CityInfo to hold City, Country, and TempAndTime.java

Removed CityWeather

michael9j2lee committed:
Changed CityInfo to include the variable names required for the api.
- id, name, countryCode, countryName, district, population, temp, time
Changed weather_show to grab from CityInfo correctly.

michael9j2lee committed:
Need to fix time zone...

michael9j2lee committed:
Modified CityInfo with default constructor and variable constructors.

Added new CityRestController with Test1 working
Added empty cityServiceTest.java

TODO:
Test2 and Test3 for CityRest controllers/CityRestControllerTest.java
Test(s) for cityServiceTest

michael9j2lee committed:
Adding New Test.

Test1 Working.  Test2 working

Todo: CityServiceTest

michael9j2lee committed:
Test 1 for CityService Test not working.
Test 2 for wrong value working.

michael9j2lee committed:
Fixed Test Cases

CarlosOrduna commented on cst438week2_assignment2/target/classes/templates/weather_show.html:
```diff
@@ -2,14 +2,43 @@
+		</tr>
+		<tr>
+			<td>Local Time</td>
+			<td>Time FIX ME</td>
```
Make sure to follow the teacher's format and show "F" after the weather

CarlosOrduna replied:
And be sure to go back and fix time

CarlosOrduna reviewed (approved):
Looks good to me. All tests seem to pass

CarlosOrduna commented on cst438week2_assignment2/src/main/java/cst438_assignment2/domain/CityInfo.java:
```diff
@@ -33,9 +29,24 @@ public CityInfo(City city, TempAndTime tempAndTime) {
-		Date date = new Date((long)(t.getTime()+t.getTimezone() )*1000);
+		Date date = new Date((long)(t.getTime() - t.getTimezone()  )*1000);
 		DateFormat format = new SimpleDateFormat("hh:mm a");
 		return format.format(date);
```
The offset should be how much you add to the getTime from getTimezone. Unfortunately I still run into the same error as well

CarlosOrduna commented on cst438week2_assignment2/src/main/java/cst438_assignment2/controllers/CityRestController.java:
```diff
@@ -24,51 +28,16 @@
 	
 	@GetMapping("/api/cities/{name}")
-	public ResponseEntity<City> cityInfo(@PathVariable("name") String name ) {
+	public ResponseEntity<CityInfo> cityInfo(@PathVariable("name") String name ) {
+		CityInfo cityInfo = cityService.getCityInfo(name);
+				
+		return new ResponseEntity<CityInfo>(cityInfo, HttpStatus.OK);
```
I have a CityInfo type class on my end, not a ResponseEntity... It works the way you did it! I would have to research more on what the difference is.

CarlosOrduna commented on cst438week2_assignment2/src/main/java/cst438_assignment2/domain/CityInfo.java:
```diff
@@ -0,0 +1,171 @@
+	
+	public CityInfo() {}
+	
+	public CityInfo(City city, TempAndTime tempAndTime) {
+		this.id = city.getID();
+		this.name = city.getName();
+		this.countryCode = city.getCountry().getCode();
+		this.countryName = city.getCountry().getName();
+		this.district = city.getDistrict();
+		this.population = city.getPopulation();
+		this.temp = tempAndTime.getTemp();
+		this.time = convertTime(tempAndTime);
+	}
```
This looks cleaner than my method. Combining both into one constructor instead of what the original file had

CarlosOrduna commented on cst438week2_assignment2/src/test/java/cst438_assignment2/domain/CityServiceTest.java:
```diff
@@ -0,0 +1,146 @@
+		assertThat(expectedResult).isEqualTo( cityService.getCityInfo( "TestCity")) ;
+		
+		verify(cityService, times(1)).getCityInfo(ArgumentMatchers.any(String.class));
+		//verify(weatherService, times(1)).getTempAndTime(ArgumentMatchers.any(String.class));
```
make sure to come back and integrate this one time is fixed

CarlosOrduna commented:
Reviewed it all and it is ready to be merged!

michael9j2lee committed:
Added F to Temperature in weather_show.html
Added default values to default constructors.
Handles empty or invalid cities and returns a small message.
